File: week-2/2-readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int letters(string text);
int words(string text);
int sentences(string text);

int main(void)
{
    string text = get_string("What's on your mind?\n");
    int count_letter = letters(text);
    int count_words = words(text);
    int count_sentences = sentences(text);

    float L = (float) count_letter / (float) count_words * 100;
    float S = (float) count_sentences / (float) count_words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    printf("Text: %s\n", text);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int letters(string text)
{
    int letters = 0;
    for (int i = 0, j = strlen(text); i < j; i++)
    {
        if (isspace(text[i]) || ispunct(text[i]) || isdigit(text[i]))
        {
            ;
        }

        else if (isupper(text[i]))
        {
            letters++;
        }

        else if (islower(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int words(string text)
{
    int words = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words;
}

int sentences(string text)
{
    int sentences = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text [i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}

File: week-6/4-sentimental-readability/readability.py
import cs50

def main():
    user_text = cs50.get_string("Text: ")

    count_letter = letters(user_text)
    count_words = words(user_text)
    count_sentences = sentences(user_text)

    L = count_letter /  count_words * 100
    S = count_sentences / count_words * 100
    coleman = round(0.0588 * L - 0.296 * S - 15.8)

    if coleman < 1:
        print("Before Grade 1")
    elif coleman >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {coleman}")


def letters(ftext):
    letters = 0

    for i in ftext:
        if i.isalpha():
            letters += 1

    return letters


def words(ftext):
    words = 1

    for i in ftext:
        if i == " ":
            words += 1

    return words


def sentences(ftext):
    sentences = 0

    for i in ftext:
        if i == "!" or i == "." or i == "?":
            sentences += 1

    return sentences


if __name__ == "__main__":
    main()
